torch/csrc/distributed/c10d/ProcessGroupNCCL.cpp

```diff
@@ -1202,8 +1216,18 @@ std::vector<std::shared_ptr<NCCLComm>>& ProcessGroupNCCL::getNCCLComm(
   ncclIdToCommMap_.emplace(buildNcclUniqueIdStr(ncclID), ncclComms);
 
   // Move the NCCL resource to cache
-  devNCCLCommMap_.emplace(devicesKey, std::move(ncclComms));
-  return devNCCLCommMap_[devicesKey];
+  auto it = inInitializationCommMap_.find(devicesKey);
```

Comment:
Is there potentially a race between `erase`  that occurs after and `find` here?

Reply:
There shouldn't be a race inside this block since the entire block is protected by the lock on line 1213.